### ISSUE TITLE
./bin/Runner.Listener: cannot execute binary file

### ISSUE THREAD

**Issue by namanrawal47** [actions/runner#2563]
**Describe the bug**
When running the config.sh script for the Linux runner image on macOS, I get `./bin/Runner.Listener: cannot execute binary file`. 

**To Reproduce**
1. Add a new runner and choose the Linux image
2. Select x64 architecture
3. Perform all auto-generated steps
4. On the following it will fail:
```
./config.sh --url https://github.boozallencsn.com/User/actions-demo --token <myToken>
```

**Expected behavior**
The runner should get created.

## Runner Version and Platform
actions-runner-linux-x64-2.296.3.tar.gz

OS of the machine running the runner? macOS Monterey 12.6.3

## What's not working?
```
$ actions-runner <myUser>$ ./config.sh --url https://github.boozallencsn.com/<myUser>/actions-demo --token <myToken>
./config.sh: line 80: ./bin/Runner.Listener: cannot execute binary file
```

## Job Log Output
See above.

**Comment by lexdevelop** [comment]
Same issue for me on ARM64 2.292.0
`github-runner-github-action-runner-1  |         not a dynamic executable
github-runner-github-action-runner-1  |         not a dynamic executable
github-runner-github-action-runner-1  |         not a dynamic executable
github-runner-github-action-runner-1  | ./config.sh: line 80: ./bin/Runner.Listener: No such file or directory
github-runner-github-action-runner-1  | /home/runner/actions-runner/run-helper.sh: line 20: /home/runner/actions-runner/bin/Runner.Listener: No such file or directory`

**Comment by koradiyakaushal** [comment]
Make sure you are using the correct CPU architecture for your action runner.

**Comment by EkantBajaj** [comment]
@lexdevelop are you able to resolve the issue ?

**Comment by lexdevelop** [comment]
> @lexdevelop are you able to resolve the issue ?
> 

Unfortunately no :(

**Comment by yaox023** [comment]
I think if you have a mac, then you should choose the macOS image. This is the solution works on my end.

**Comment by Tienisto** [comment]
I get the same error with `runner-linux-arm-2.311.0` on my arm64 server

**Comment by dcuccia** [comment]
Fixed it for me - was running x64 on ARM64. 🤦‍♂️

**Comment by Tienisto** [comment]
Yes! I missed the "64" part and only selected "ARM" thinking that 64 bit is default

**Comment by onmyway133** [comment]
If you are on M (sillicon) Mac, make sure you select macOS and ARM64

<img width="349" height="197" alt="Image" src="https://github.com/user-attachments/assets/4dfb285f-30b7-4748-befe-50de4cea06e1" />
